### PR TITLE
Integrate Jenkins CICD 

### PR DESCRIPTION
**Pull request by SagiPolaczek**
# Ready for CR 🕺🏼

### Remarks
1. Couldn't solve the annoying error: 
```
Failed to read test report file ...../test-reports/TEST-fusedrug_examples.tests.test_interaction_cat_dti.InterCatDtiTestCase-20230323060652.xml
org.dom4j.DocumentException: Error on line 114 of document  : An invalid XML character (Unicode: 0x1b) was found in the CDATA section.
```
Somehow Lightning's logger prints a special unicode character (0x1b) that Jenkins can't parse. I added a workaround for that but it is something that we should figure out (either in this PR or in a future one).

2. The original submodule repository is not stable so I switched to @alex-golts 's fork which should be stable.
3. `fusedrug/utils/file_formats/tests/test_indexed_fasta_custom.py` crashed due some changes in #20. I added a workaround but generally speaking I don't like that the output changes its structure (once a string and once a tuple). What do you think @YoelShoshan  ?

### CODE REVIEW

File: fusedrug/utils/file_formats/tests/test_indexed_fasta_custom.py [hunk 1]
```diff
@@ -56,6 +57,19 @@ def test_identifier_based_access(self):
 
         self.assertTupleEqual(ans_1, ans_2)
 
+    def get_raw_identifier(self, identifier: Union[str, Tuple[str, str]]) -> str:
+        """
+        An helper function to (maybe temporarily) address Yoel's changes in PR #20 at:
+        fusedrug_public/fusedrug/utils/file_formats/indexed_fasta_custom.py line 27~
+
+        TODO double check with Yoel
```
Comment (SagiPolaczek) [inline fusedrug/utils/file_formats/tests/test_indexed_fasta_custom.py:65]: See [here](https://github.com/BiomedSciAI/fuse-drug/pull/20/files#diff-de86e19ca5200e0550ef170e5e839a083016fb21e19eaabb7cb802a8a6c5cd3d)

File: fusedrug_examples/interaction/drug_target/affinity_prediction/PLM_DTI/plm_dti.py [hunk 2]
```diff
@@ -62,7 +62,7 @@ def validation_step(self, batch_dict: NDict, batch_idx: int) -> None:
         batch_dict = self.model(batch_dict)
         loss = fuse_pl.step_losses(self.losses, batch_dict)
         for _, met_instance in self.val_metric_dict.items():
-            met_instance(batch_dict["model.output"], batch_dict["data.label"])
+            met_instance(batch_dict["model.output"], batch_dict["data.label"].type(torch.int))
```
Comment (SagiPolaczek) [inline fusedrug_examples/interaction/drug_target/affinity_prediction/PLM_DTI/plm_dti.py:65]: match types (torchmetrics err)

File: run_all_unit_tests.py [hunk 3]
```diff
@@ -3,61 +3,79 @@
...
-        # TODO: delete the created cache
-        cache_path = caching.get_current_read_locations(caching_kind="user_local")
-        assert 1 == len(cache_path)
-        cache_path = cache_path[0]
-        assert "ms_img_analytics" in cache_path
-        # delete the cache that we've created for this job
-        print(f"deleting eir cache {cache_path} ...")
-        os.system(f"rm -rf {cache_path}")
-        print("done deleting eir cache.")
+    ### A workaround for "An invalid XML character" issue for the examples' unit-tests
```
Comment (SagiPolaczek) [inline run_all_unit_tests.py:64]: Currently couldn't find the origin for that special character...

File: requirements/requirements.txt [hunk 4]
```diff
@@ -2,6 +2,7 @@ click
 pyfastx
 torch
 pytorch-lightning<2.0.0 # temp - need to make all tests pass with this version first
+torchvision<0.15  # temp
```
Comment (mosheraboh) [inline requirements/requirements.txt:5]: still required? add a comment that explains the issue so we will be able to revert.
Also create a trello card to resolve this.

File: run_all_unit_tests.sh [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+            # install PyTorch
+            if [ $force_cuda_version != "no" ]; then
+                echo "forcing cudatoolkit $force_cuda_version"
+                conda install $env pytorch torchvision pytorch-cuda=$force_cuda_version -c pytorch -c nvidia
+                echo "forcing cudatoolkit $force_cuda_version - Done"
+            fi
+
+            echo "Installing FuseMedML"
+            conda run $env --no-capture-output --live-stream pip install git+https://github.com/BiomedSciAI/fuse-med-ml@master
```
Comment (mosheraboh) [inline run_all_unit_tests.sh:73]: How do we force to bump fuse version? Change the requirement file?

**Review by mosheraboh — APPROVED** [review]
Thanks Sagi!
Looks good.
Two minors inline.

File: run_all_unit_tests.sh (hunk 5 above)
Comment (SagiPolaczek) [reply run_all_unit_tests.sh:73]: With that being said, if the conda env already exists we will NOT update fuse.

File: requirements/requirements.txt (hunk 4 above)
Comment (SagiPolaczek) [reply requirements/requirements.txt]: Nice catch. I'm checking it now :)

File: run_all_unit_tests.sh (hunk 5 above)
Comment (mosheraboh) [reply run_all_unit_tests.sh:73]: Yes, I was asking about the case it already exists.

**Review by mosheraboh — APPROVED** [review]
Looks good; you can go ahead and merge.

**Pull request merged**